dynamic/dynamic_message.go

```diff
@@ -1921,44 +1921,44 @@ func validElementFieldValue(fd *desc.FieldDescriptor, val interface{}) (interfac
 
 func validElementFieldValueForRv(fd *desc.FieldDescriptor, val reflect.Value) (interface{}, error) {
 	t := fd.GetType()
-	typeName := strings.ToLower(t.String())
 	if !val.IsValid() {
-		return nil, fmt.Errorf("%s field %s is not compatible with nil value", typeName, fd.GetFullyQualifiedName())
+		return nil, typeError(fd, val.Type())
```

Comment:
This will panic like so:
```
panic: reflect: call of reflect.Value.Type on zero Value
```
I think you can instead just pass `nil` for the type, and it will come out as `"<nil>"` in the resulting error message (which I think is fine).

Reply:
Good catch! fixed